docs/develop/scripting-fundamentals.md

```diff
@@ -218,6 +218,120 @@ function main(workbook: ExcelScript.Workbook) {
 }
 ```
 
+### Definining types 
+
+Type declarations help users understand the type of variable they are dealing with. It helps with auto completion of methods and assists in development time quality checks. 
+You can fine type declarations in the script in various places including function declaration, variable declaration, IntelliSence definitions, etc. 
+
+Examples: 
+
+* `function main(workbook: ExcelScript.Workbook)`
+* `let myRange: ExcelScript.Range;` 
+* `function getMaxAmount(range: ExcelScript.Range): number`
+
+You can identify the types easily in the code editor as it appears distinctly usually in a different color. The color `:` usually preceds the type declaration.  
+
+Writing types can be optional in TypeScript, because type inference allows you to get a lot of power without writing additional code. For the most part TypeScript language is good at inferring the types of variables. However, in certain cases, Office Script will require the type declarations be explicitly defined if the language is unable to clearly identify the type. Also, explicit or implicit `any` is not allowed in Office Script. More on that below 
+
+#### 'any' type in the script
+
+As the [TypeScript website states](https://www.typescriptlang.org/docs/handbook/basic-types.html#any), in some situations, not all type information is available or its declaration would take an inappropriate amount of effort. These may occur for values from code that has been written without TypeScript or a 3rd party library. In these cases, we might want to opt-out of type checking. To do so, we label these values with the `any` type:
+
+```typescript
+declare function getValue(key: string): any;
+// OK, return value of 'getValue' is not checked
+const str: string = getValue("myString");
+```
+
+**Explicit `any` is not allowed**
+
+```typescript
+let someVariable: any; 
+// ^^ This is not allowed ^^
```

Comment:
@Rick-Kirkham, I remember our team, as some point, had a discussion about how to present bad code as an example. Do you remember the formatting we landed on?